src/storage/heap_file.c

```diff
@@ -25065,4 +25090,48 @@ heap_update_and_log_header (THREAD_ENTRY * thread_p, const HFID * hfid, const PG
   return NO_ERROR;
 }
 
+void
+heap_log_postpone_heap_append (THREAD_ENTRY * thread_p, const HFID * hfid, const OID * class_oid,
+                               const std::vector <VPID> heap_pages_array)
+{
+  // This append needs to be run on postpone after the commit.
+  // First create the log data required.
+  size_t array_size = heap_pages_array.size ();
+  size_t log_data_size = sizeof (HFID) + sizeof (OID) + sizeof (size_t) + array_size * sizeof (VPID);
+  char *log_data_buffer = (char *) db_private_alloc (NULL, log_data_size + MAX_ALIGNMENT);
+  char *log_data = PTR_ALIGN (log_data_buffer, MAX_ALIGNMENT);
+  LOG_DATA_ADDR log_addr = LOG_DATA_ADDR_INITIALIZER;
+
+  // Now populate the log data needed.
+  size_t offset = 0;
+
+  // HFID
+  HFID_COPY ((HFID *) (log_data + offset), hfid);
+  offset += sizeof (HFID);
+
+  // class_oid
+  COPY_OID ((OID *) (log_data + offset), class_oid);
```

Comment:
To copy through pointer, 8 bytes alignment is required. 
please notice that sizeof (HFID) == 12. 

You may have 2 choices:
1. use OR_PUT_PID instead: it requires INT_ALIGNMENT
2. change like this:
```c
  ptr = log_data;

  // copy HFID
  ptr += sizeof (HFID);

  ptr = PTR_ALIGN (ptr, PTR_ALIGNMENT);
  COPY_OID ;
  ptr += sizeof (OID);

  ...
```